java/src/jni/h5aImp.c

```diff
@@ -158,26 +158,29 @@ Java_hdf_hdf5lib_H5_H5Aread(JNIEnv *env, jclass clss, jlong attr_id, jlong mem_t
 {
     jboolean readBufIsCopy;
     jbyte *  readBuf = NULL;
-    htri_t   data_class;
+    hsize_t  dims[H5S_MAX_RANK];
+    hid_t    sid = H5I_INVALID_HID;
+    jsize    n;
+    htri_t   vl_data_class;
     herr_t   status = FAIL;
 
     UNUSED(clss);
 
     if (NULL == buf)
         H5_NULL_ARGUMENT_ERROR(ENVONLY, "H5Aread: read buffer is NULL");
 
-    if ((data_class = H5Tdetect_class(mem_type_id, H5T_VLEN)) < 0)
-        H5_LIBRARY_ERROR(ENVONLY);
-
-    if (data_class)
-        H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Aread: variable length type not supported");
+    /* Get size of data array */
+    if ((n = ENVPTR->GetArrayLength(ENVONLY, buf)) < 0) {
+        CHECK_JNI_EXCEPTION(ENVONLY, JNI_TRUE);
+        H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Aread: readBuf length < 0");
+    }
 
-    /* Recursively detect any vlen string in type (compound, array ...) */
-    if ((data_class = H5Tdetect_variable_str(mem_type_id)) < 0)
+    dims[0] = (hsize_t)n;
+    if ((sid = H5Screate_simple(1, dims, NULL)) < 0)
```

Comment:
I would probably only create this space in the cleanup block if an H5Treclaim is necessary, but that's a minor nitpick.

Reply:
Maybe - I'll check.

Reply:
Fixed.